schemes/musica/tuvx/musica_ccpp_tuvx_cloud_optics.F90

```diff
@@ -0,0 +1,128 @@
+! Copyright (C) 2024 National Science Foundation-National Center for Atmospheric Research
+! SPDX-License-Identifier: Apache-2.0
+module musica_ccpp_tuvx_cloud_optics
+  implicit none
+
+  private
+  public :: create_cloud_optics_radiator, set_cloud_optics_values
+
+  ! This module is used to set the optical properties of clouds in TUV-x.
+  ! Optical properties are defined as a function of wavelength and height,
+  ! and include the cloud optical depth, single scattering albedo,
+  ! and asymmetry parameter.
+  !
+  ! See module_ccpp_tuvx_height_grid for the definition of the height grid
+  ! and its mapping to the CAM-SIMA vertical grid.
+
+  !> Label for cloud optical properties in TUV-x
+  character(len=*), parameter, public :: cloud_optics_label = "clouds"
+  !> Default value of number of vertical levels
+  integer, parameter :: DEFAULT_NUM_VERTICAL_LEVELS = 0
+  !> Number of vertical levels
+  integer, protected :: num_vertical_levels = DEFAULT_NUM_VERTICAL_LEVELS
+  !> Default value of number of wavelength bins
+  integer, parameter :: DEFAULT_NUM_WAVELENGTH_BINS = 0
+  !> Number of wavelength bins
+  integer, protected :: num_wavelength_bins = DEFAULT_NUM_WAVELENGTH_BINS
+
+contains
+
+  !> Creates a TUV-x cloud optics radiator from the host-model wavelength grid
+  function create_cloud_optics_radiator( height_grid, wavelength_grid, &
+      errmsg, errcode ) result( radiator )
+    use musica_ccpp_util,     only: has_error_occurred
+    use musica_tuvx_grid,     only: grid_t
+    use musica_tuvx_radiator, only: radiator_t
+    use musica_util,          only: error_t
+
+    type(grid_t),                        intent(inout) :: height_grid
+    type(grid_t),                        intent(inout) :: wavelength_grid
+    character(len=*),                    intent(out)   :: errmsg
+    integer,                             intent(out)   :: errcode
```

Comment:
Is the large spacing intentional?

Reply:
not by me. copilot may have had a reason, but I've removed the extra spaces.